webhook/resources/volume/mutator.go

```diff
@@ -97,7 +97,24 @@ func (v *volumeMutator) Create(request *admission.Request, newObj runtime.Object
 	}
 
 	if string(volume.Spec.AccessMode) == "" {
-		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/accessMode", "value": "%s"}`, string(longhorn.AccessModeReadWriteOnce)))
+		accessModeFromBackup := longhorn.AccessModeReadWriteOnce
+		if volume.Spec.FromBackup != "" {
+			bName, _, _, err := backupstore.DecodeBackupURL(volume.Spec.FromBackup)
+			if err != nil {
+				return nil, werror.NewInvalidError(fmt.Sprintf("cannot get backup and volume name from backup URL %v: %v", volume.Spec.FromBackup, err), "")
+			}
+			bs, err := v.ds.GetBackupRO(bName)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to list backups for volumes: %v", bName)
```

Comment:
Why not use werror?

Reply:
I did not follow consistent to use `werror`, I will correct it as `werror.NewInternalError` to return exception information.